### PR TITLE
Create StrobogrammaticNumber.java

### PR DESCRIPTION
**Pull request by HimeshKohad**
- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] All filenames are in PascalCase.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.

### CODE REVIEW

**Review by HimeshKohad — COMMENTED** [review]
tried fixing linter issues

File: src/main/java/com/thealgorithms/maths/StrobogrammaticNumber.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+            if (!strobogrammaticMap.containsKey(leftChar) || strobogrammaticMap.get(leftChar) != rightChar) {
+                return false;
+            }
+
+            left++;
+            right--;
+        }
+
+        return true;
```
Comment (madhavgaba) [inline src/main/java/com/thealgorithms/maths/StrobogrammaticNumber.java:41]: add tests

Comment (madhavgaba) [reply src/main/java/com/thealgorithms/maths/StrobogrammaticNumber.java:41]: yes, please add it here.

Comment (HimeshKohad) [reply src/main/java/com/thealgorithms/maths/StrobogrammaticNumber.java:41]: We are not supposed to use the main method right? Can you suggest how to implement tests without the main method?

**Review by siriak — COMMENTED** [review]
Could you please add some JUnit tests? (see https://github.com/TheAlgorithms/Java/tree/master/src/test/java/com/thealgorithms)

**Comment by HimeshKohad** [comment]
> Could you please add some JUnit tests? (see https://github.com/TheAlgorithms/Java/tree/master/src/test/java/com/thealgorithms)

I have created PR #4273 StrobogrammaticNumberTest. Can you help me with its build failure?

**Comment by siriak** [comment]
It says Error:  /home/runner/work/Java/Java/src/test/java/com/thealgorithms/maths/StrobogrammaticNumberTest.java:[11,9] cannot find symbol
  symbol:   class StrobogrammaticNumber
  location: class com.thealgorithms.maths.StrobogrammaticNumberTest
  Because tests and code must be in the same PR

**Comment by siriak** [comment]
https://github.com/TheAlgorithms/Java/actions/runs/5717366238/job/15490947507?pr=4273

**Comment by HimeshKohad** [comment]
> It says Error: /home/runner/work/Java/Java/src/test/java/com/thealgorithms/maths/StrobogrammaticNumberTest.java:[11,9] cannot find symbol symbol: class StrobogrammaticNumber location: class com.thealgorithms.maths.StrobogrammaticNumberTest Because tests and code must be in the same PR

How can I make a single PR for test and code?

**Comment by siriak** [comment]
Commit them to the same branch and open a PR for that branch

**Comment by debasishbsws** [comment]
If you already raised a PR, then you can just add the test to the same branch, commit and push it, and it will be reflected in the same PR.

**Comment by HimeshKohad** [comment]
> If you already raised a PR, then you can just add the test to the same branch, commit and push it, and it will be reflected in the same PR.

Thank you for the help!

**Review by HimeshKohad — COMMENTED** [review]
tried fixing build error

**Review by siriak — APPROVED** [review]
Looks good, thanks!

**Pull request merged**